pkg/virt-launcher/virtwrap/converter/arm64.go

```diff
@@ -0,0 +1,59 @@
+/* Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright the KubeVirt Authors.
+ *
+ */
+
+package converter
+
+import (
+	v1 "kubevirt.io/api/core/v1"
+
+	"kubevirt.io/kubevirt/pkg/pointer"
+	"kubevirt.io/kubevirt/pkg/virt-launcher/virtwrap/api"
+)
+
+type archConverterARM64 struct{}
+
```

Comment:
Can you please add a `var _ = archConverter(&archConverterARM64{})`?
This will ensure in compilation time that the struct implements the interface and fail otherwise.
Please do so for all 3 converters.

Reply:
The `newArchConverter` function already explicitly returns `archConverter`, so this is already covered.
For example if you comment out isUSBNeeded:
```
pkg/virt-launcher/virtwrap/converter/archconverter.go:40:10: cannot use archConverterARM64{} (value of type archConverterARM64) as archConverter value in return statement: archConverterARM64 does not implement archConverter (missing method isUSBNeeded)
```

Reply:
I see your point.

My problem with it is that it's implicit and hard to understand for someone not familiar enough with the code. For example, when looking at this file, how can I know that it aims to implement an interface? In this case it's even more important, as this struct's whole purpose (at least currently) is to implement this interface.

I think that adding it here will be valuable as it will make the code more cohesive, clearer and self-contained. But I'm willing to be convinced otherwise.

Is there a reason you think it's more valuable to not have this validation here? Are there any downsides to it?

Reply:
No that makes sense.
The only real downside i see would be people wondering why the line is there. But i have just added a comment to it for that.